### ISSUE TITLE
BUG: Acode using Samsung keyboard overwrites words when "Keyboard mode" is "Normal" or "No suggestion"

### ISSUE THREAD

**Issue by oskint** [Acode-Foundation/Acode#1057]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

(I firstly did Settings >> Restore default settings.)
Use Samsung keyboard.  (Edited after comments.)

Go to
Settings
  App Settings
    Keyboard mode
      and select:
        Normal or
        No suggestion

In editor go to the new line
and write e.g.
aaaa bbb
Then click e.g. in the end of the word aaaa
and insert space.

RESULT:
Acode automatically changes text
aaaa bbb
to 
abbb bbb

It means it overwrites word aaaa with bbb.

The bug is so annoying that this menu option is completely unusable.
And without the option user is forced manually correct every typo. In my case to use other editor.

### Environment

Samsung A52
Samsung keyboard application 5.6.00.51
Android 13
Acode 1.10.5
20th Oct 2024

### If applicable, attach your Acode.log file to this issue.

<details><summary>Acode.log</summary><pre>

</pre></details>

**Comment by bajrangCoder** [comment]
I tried to reproduce it. But failed.

As it's working fine for me with all the three keyboard modes available.(Btw the no suggestions mode is good for different devices, as it doesn't disable suggestions)
I'm using gboard.

**Comment by bajrangCoder** [comment]
Maybe your have any ai based , system level auto correct feature, which can cause this

I guess you are using samsung keyboard, ?

**Comment by oskint** [comment]
I use Samsung keyboard version 5.6.00.51 (just actualised) and the bug is still there. 
But you are right, using the Gboard the bug disappears, I just tried it. 
E.g. QuickEdit editor with Samsung keyboard does not produce the bug.
Therefore the cause is the combination of Acode with Samsung keyboard. 

Thank you. 
I edited the title.

**Comment by NishchayKQ** [comment]
on fresh install, acode 1.10.5(953), samsung keyboard - 5.9.00.63, one ui 6.1.1,
normal keyboard mode/no suggestions, i couldn't reproduce it

tried on an older device as well (samsung keyboard 5.6.10.40)

**Comment by oskint** [comment]
My problem persists 
even after updates, clearing data and caches, and removing languages.

------ Just updated my Samsung Galaxy A52:

Android 14  
(writes "Your software is up to date"), English (US), removed other languages.
Settings >> General Management >> Language: English (United States)

Samsung One UI 6.1.

Samsung Keyboard 5.8.20.10 
(writes "The latest version installed"), cleared cache & data.
Languages and types: English (US), removed other language
Auto replace: no languages selected

Acode 1.10.5
just uninstalled + installed, cleared cache & data
Settings >> App Settings >> Keyboard mode: Normal

------ BUG REPRODUCING 
Change happened.

On a new line typing 
aaaa bbb
then moving to the end of "aaaa" and typing SPACE
does nothing - works OK.

But,
when i 
move the cursor to the end of "aaaa" 
and write e.g. letter c
then the result is
abbbc bbb

That is
instead of result
aaaac bbb
it rewrites it to
abbbc bbb.

SO, THE MODIFIED BUG IS STILL THERE.

OTHER APPS:
QuickEdit editor: no such bug
Jota+ editor: no such bug
NeutriNote editor: no such bug
WhatsApp: message input: no such bug

--------
I am really furious about the Acode thing
because in hunt for the bug solution
I had to delete many settings that worked for me ok for years.

--------
Samsung Galaxy A52
Android 14 - English (United States)
Samsung One UI 6.1
Samsung Keyboard - 5.8.20.10 - English (US)
Acode 1.10.5
--------
Any ideas?.
Thank you in advance

**Comment by NishchayKQ** [comment]
yes the modified version of the bug i can consistently reproduce, further the old bug still is there, its just not consistent to reproduce it

https://github.com/user-attachments/assets/637ab4af-b8e3-4e37-98f4-abf2cb5c32f0

**Comment by UnschooledGamer** [comment]
I've tested it with Latest Version of Acode(Play Store) on my samsung device with samsung keyboard, Confirm it as a Bug. It's a consistent bug for me.

**Comment by UnschooledGamer** [comment]
Update:
I turned off predictive text from samsung keyboard settings, And now the Acode typing experience feels fine.

**Comment by RohitKushvaha01** [comment]
@bajrangCoder I don’t have a Samsung device, so I’m not sure I’ll be able to fix this.

**Comment by deadlyjack** [comment]
You can use online emulator for samsung device to test.

**Comment by NishchayKQ** [comment]
turning predictive text off does fix the issue.

However if i have it on, and disable auto replace, it still results in the bug
(ex: use case suggestions are allowed but don't automatically replace)
